tests/preserve-encodings/input.cddl

```diff
@@ -111,4 +111,33 @@ bounds_group_choice = [
     hash //
     ; @name c
     1, x: hash, y: hash
+]
+
+; just testing this compiles (for now? possibly have full tests in a later commit)
```

Comment:
Is this comment meant to be merged in as-is?

Reply:
I wasn't sure if it was worth it to put in preserve-encodings tests since all the deser code should be tested in the other tests (the base group choice one + other tests in general) so the big worry was just about it compiling. I guess it can't hurt to have better coverage though so I added two new cases for those two definitions in the last commit.